aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -611,19 +605,38 @@ TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tenso
 void TensorIterator::mark_outputs() {
   for (int i = 0; i < num_outputs_; i++) {
     operands_[i].is_output = true;
-    const auto &output = operands_[i].tensor;
+    const auto& output = operands_[i].tensor;
     if (!output.defined()) continue;
 
     // check if output is also an input
     for (int arg = num_outputs_; arg < ntensors(); arg++) {
-      const auto &input = operands_[arg].tensor;
+      const auto& input = operands_[arg].tensor;
       if (output.is_same(input)) {
         operands_[i].is_read_write = true;
       }
     }
   }
 }
 
+void TensorIterator::check_mem_overlaps() {
+  if (check_mem_overlap_) {
```

Comment:
Nit: this is just a style choice, but sometimes I think it is nicer to write
```
if (!check_mem_overlap_) {
  return;
}
// the rest of the fn
```
to avoid the extra indentation. But it's totally fine either way

Reply:
addressed